cost-analyzer/values.yaml

```diff
@@ -996,78 +974,61 @@ kubecostDeployment:
-      # seccompProfile:
-      #   type: RuntimeDefault
-    containerSecurityContext:
-      allowPrivilegeEscalation: true
-      readOnlyRootFilesystem: false
-      capabilities:
-        drop:
-        - ALL
-    resources:
-      requests:
-        ## You can use the Kubecost savings report for 'Right-size your
-        ## container requests' to determine the recommended resource requests
-        ## once the pod has run for 24 hours.
-        cpu: 1000m
-        memory: 500Mi
-    ## default storage class
-    storageClass: ""
-    databaseVolumeSize: 100Gi
-    configVolumeSize: 1Gi
-    initImage: {}
-    # Optional - add extra ports to the query service container. For kubecost development purposes only - not recommended for users.
-    extraPorts: []
-      # - name: debug
-      #   port: 40000
-      #   targetPort: 40000
-      #   containerPort: 40000
-
 ## The Kubecost Aggregator is a high scale implementation of Kubecost intended
 ## for large datasets and/or high query load. At present, this should only be
 ## enabled when recommended by Kubecost staff.
 ##
 kubecostAggregator:
+  # deployMethod determines how Aggregator is deployed. Current options are
+  # "singlepod" (within cost-analyzer Pod) "statefulset" (separate
+  # StatefulSet), and "disabled". Only use "disabled" if this is a secondary
+  # Federated ETL cluster which does not need to answer queries.
+  deployMethod: "singlepod"
+
+  # For legacy configuration support, `enabled: true` overrides deployMethod
+  # and causes `deployMethod: "statefulset"`
   enabled: false
+
+  # Replicas sets the number of Aggregator replicas. It only has an effect if
+  # `deployMethod: "statefulset"`
   replicas: 1
-  ## Creates a new pod to retrieve CloudCost data. By default it uses the same
-  ## serviceaccount as the cost-analyzer pod. A custom serviceaccount can be
-  ## specified.
+
+  ## Creates a new container/pod to retrieve CloudCost data. By default it uses
+  ## the same serviceaccount as the cost-analyzer pod. A custom serviceaccount
+  ## can be specified.
   cloudCost:
+    # The cloudCost component of Aggregator depends on
+    # kubecostAggregator.deployMethod:
+    # kA.dM = "singlepod" -> cloudCost is run as container inside cost-analyzer
+    # kA.dM = "statefulset" -> cloudCost is run as single-replica Deployment
     enabled: false
     # serviceAccountName:
+
+  # Jaeger is an optional container attached to wherever the Aggregator
+  # container is running. It is used for performance investigation. Enable if
+  # Kubecost Support asks.
   jaeger:
     enabled: false
     image: jaegertracing/all-in-one
     imageVersion: latest
     # containerSecurityContext:
   # fullImageName:
-  resources: {}
+  resources:
+    # TODO: Benchmark requests to determine minimal
+    # stable set.
+    requests:
+      cpu: "1000m"
+      memory: "2Gi"
   env:
     "LOG_LEVEL": "info"
+    "DB_READ_THREADS": "1"
+    "DB_WRITE_THREADS": "1"
+    "DB_MEMORY_LIMIT": "500MB"
```

Comment:
I would leave unset and let duck use its 'percentage of total ram' default logic

Reply:
I'm worried about this because of the way DuckDB treats CPU and memory. We know that letting DuckDB pick its own CPU value is bad because it picks node max and then we get instability (I've seen crashes that are mitigated just by setting DB_READ_THREADS < max CPU). I'm also in favor of constraining memory by default because we've seen DuckDB "keep" memory allocated (we're still trying to figure out why as you know) and I'd rather than amount stay very low.

Reply:
very good!